src/bundle/ui-dev/src/modules/common/popup-menu/popup.menu.js

```diff
@@ -0,0 +1,194 @@
+import React, { useState, useEffect, useRef, useMemo } from 'react';
+import PropTypes from 'prop-types';
+
+import { getTranslator, getRootDOMElement } from '@ibexa-admin-ui/src/bundle/Resources/public/js/scripts/helpers/context.helper';
+import { createCssClassNames } from '@ibexa-admin-ui/src/bundle/ui-dev/src/modules/common/helpers/css.class.names';
+import Icon from '@ibexa-admin-ui/src/bundle/ui-dev/src/modules/common/icon/icon';
+
+const MIN_SEARCH_ITEMS_DEFAULT = 5;
+const MIN_ITEMS_LIST_HEIGHT = 150;
+
+const PopupMenu = ({ extraClasses, footer, items, onItemClick, positionOffset, referenceElement, scrollContainer, onClose }) => {
+    const Translator = getTranslator();
+    const containerRef = useRef();
+    const [isRendered, setIsRendered] = useState(false);
+    const [itemsListStyles, setItemsListStyles] = useState({
+        left: 0,
+        top: 0,
+    });
+    const [filterText, setFilterText] = useState('');
+    const numberOfItems = useMemo(() => items.reduce((sum, group) => sum + group.items.length, 0), [items]);
+    const popupMenuClassName = createCssClassNames({
+        'c-popup-menu': true,
+        'c-popup-menu--hidden': !isRendered,
+        [extraClasses]: true,
+    });
+    const searchPlaceholder = Translator.trans(/*@Desc("Search...")*/ 'ibexa_popup_menu.search.placeholder', {}, 'ibexa_popup_menu');
+    const updateFilterValue = (event) => setFilterText(event.target.value);
+    const resetInputValue = () => setFilterText('');
+    const showItem = (item) => {
+        if (filterText.length < 3) {
+            return true;
+        }
+
+        const itemLabelLowerCase = item.label.toLowerCase();
+        const filterTextLowerCase = filterText.toLowerCase();
+
+        return itemLabelLowerCase.indexOf(filterTextLowerCase) === 0;
+    };
+    const renderGroup = (group) => {
+        const isAnyItemVisible = group.items.some(showItem);
+
+        if (!isAnyItemVisible) {
+            return null;
+        }
+
+        const groupClassName = createCssClassNames({
```

Comment:
I'm not sure if we should use `createCssClassNames` because this it's only one class and it's not conditional